aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/mobileconnectors/kinesisvideo/auth/KinesisVideoCredentialsProviderImpl.java

```diff
@@ -67,6 +69,15 @@ protected KinesisVideoCredentials updateCredentials() throws KinesisVideoExcepti
 
             expiration = cognitoCredentialsProvider.getSessionCredentialsExpiration();
             log.debug("Refreshed token expiration is %s", expiration);
+        } else if (credentialsProvider instanceof AWSMobileClient) {
+            AWSMobileClient awsMobileClient = (AWSMobileClient) credentialsProvider;
+            try {
```

Comment:
What happens to expiration time when credentials provider is none of these checked types?

Reply:
The expiration date is then left at what it was previously: `KinesisVideoCredentials.CREDENTIALS_NEVER_EXPIRE` (`new Date(Long.MAX_VALUE)`), so this `updateCredentials()` method will basically never execute a second time.